terraform/modules/swipe-sfn-batch-job/main.tf

```diff
@@ -34,6 +34,7 @@ locals {
     "MINIWDL__DOWNLOAD_CACHE__DISABLE_PATTERNS" = "[\"s3://swipe-samples-*/*\"]",
     "DOWNLOAD_CACHE_MAX_GB"                     = "500",
     "WDL_PASSTHRU_ENVVARS"                      = join(" ", [for k, v in var.extra_env_vars : k]),
+    "AWS_STEP_NOTIFICATION_PLUGIN"              = var.sfn_notification_queue_urls[0],
```

Comment:
Using `[0]` here kind of breaks the idea of multiple queues. The first queue gets all these messages but it is not obvious that it is special. The reason for multiple queues is because processing events consumes them so we might want multiple listeners on these events.

Instead of calling SQS multiple times from the plugin I think a better fit would be SNS. Right now, EventBridge gives us an SNS notification then we subscribe that SNS notification from each of the SQS queues, which handles the fan out. We already have a [topic](https://github.com/chanzuckerberg/swipe/blob/main/terraform/modules/swipe-sfn/notifications.tf#L34C1-L39C2) hooked up to these queues so using that here should give you access to all of them in one call. 

```hcl
resource "aws_sns_topic" "sfn_notifications_topic" {
  count = local.enable_notifications ? 1 : 0

  name = "${var.app_name}-sfn-notifications-topic"
  tags = var.tags
}

# ...

resource "aws_sns_topic_subscription" "sfn_notifications_sqs_target" {
  for_each = var.sqs_queues

  topic_arn = aws_sns_topic.sfn_notifications_topic[0].arn
  protocol  = "sqs"
  endpoint  = aws_sqs_queue.sfn_notifications_queue[each.key].arn
}
```

You can publish to SNS similarly to how you can to SQS (also would prefer to change the name to something a bit more descriptive like `STEP_NOTIFICATION_TOPIC_ARN`).

```python
sns = boto3.client('sns')

topic_arn = os.getenv('STEP_NOTIFICATION_TOPIC_ARN')

# Send the message
response = sns.publish(
    TopicArn=topic_arn,
    Message='Hello, this is a test message from boto3!',
    Subject='Test Subject'
)

# Print out the message ID
print(response['MessageId'])
```